After recent merge, pnetcdf build no longer working

In my CI, the pnetcdf build just broke:

```
ncpdispatch.c: In function ‘NCP_inq_var_all’:
ncpdispatch.c:1176:8: error: ‘options_maskp’ undeclared (first use in this function)
     if(options_maskp) return NC_ENOTNC4;
        ^~~~~~~~~~~~~
ncpdispatch.c:1176:8: note: each undeclared identifier is reported only once for each function it appears in
ncpdispatch.c: At top level:
ncpdispatch.c:1596:1: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
 NCP_set_var_chunk_cache,
 ^~~~~~~~~~~~~~~~~~~~~~~
ncpdispatch.c:1596:1: note: (near initialization for ‘NCP_dispatcher.def_var_filter’)
ncpdispatch.c:1597:1: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
 NCP_get_var_chunk_cache,
 ^~~~~~~~~~~~~~~~~~~~~~~

```

When I look at the code, it seems like it could never have compiled. options_maskp is not declared, but is used in the code.

@WardF do you think you have a travis pnetcdf build? If so, can you check to see why it didn't catch this?

--- a/libsrcp/ncpdispatch.c
+++ b/libsrcp/ncpdispatch.c
@@ -1173,7 +1173,6 @@ NCP_inq_var_all(int ncid, int varid, char *name, nc_type *xtypep,
     if(contiguousp) *contiguousp = NC_CONTIGUOUS;
     if(no_fill) *no_fill = 1;
     if(endiannessp) return NC_ENOTNC4;
-    if(options_maskp) return NC_ENOTNC4;
     return NC_NOERR;
 }
 
@@ -1502,6 +1501,12 @@ NCP_def_var_endian(int ncid, int varid, int endianness)
     return NC_ENOTNC4;
 }
 
+static int
+NCP_def_var_filter(int ncid, int varid, unsigned int id, size_t nparams, const unsigned int* parms)
+{
+    return NC_ENOTNC4;
+}
+
 #endif /*USE_NETCDF4*/
 
 /**************************************************/
@@ -1593,6 +1598,7 @@ NCP_def_var_fletcher32,
 NCP_def_var_chunking,
 NCP_def_var_fill,
 NCP_def_var_endian,
+NCP_def_var_filter,
 NCP_set_var_chunk_cache,
 NCP_get_var_chunk_cache,
 #endif /*USE_NETCDF4*/

--- a/libsrcp/ncpdispatch.c
+++ b/libsrcp/ncpdispatch.c
@@ -1173,6 +1173,9 @@ NCP_inq_var_all(int ncid, int varid, char *name, nc_type *xtypep,
     if(contiguousp) *contiguousp = NC_CONTIGUOUS;
     if(no_fill) *no_fill = 1;
     if(endiannessp) return NC_ENOTNC4;
+    if(idp) return NC_ENOTNC4;
+    if(nparamsp) return NC_ENOTNC4;
+    if(params) return NC_ENOTNC4;
     return NC_NOERR;
 }
 

